### PR TITLE
fix: restore merging of hits in TrackerHitCollector

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This restores merging of all tracker hits in the TrackerHitCollector. It does not assume that all collections are present but empty if not supported by the geometry configuration. Instead, it catches the exception on JEvent::GetCollectionBase if a podio collection is requested that does not exist, and continues with the other (existing) podio collections. The transition to podio collections upstream, in

This also reworks the factory to the new multifactory approach, moves it into src/factories, and uses the modular algorithms for the operations on podio collections instead of replicating the functionality in the factory.

This addresses https://github.com/eic/EICrecon/pull/840#issuecomment-1692340071, bullet 3.

### What kind of change does this PR introduce?
- [x] Bug fix (issue: #840 introduced a regression by removing all hits from CentralTrackingRecHits)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
Yes, ensures CentralTrackingRecHits are filled again.

### CODE REVIEW

File: src/global/tracking/tracking.cc [hunk 1]
```diff
@@ -33,22 +34,23 @@ void InitPlugin(JApplication *app) {
...
+            "TOFEndcapRecHits",
+            "MPGDBarrelRecHits",           // MPGD
+            "MPGDDIRCRecHits",
+            "OuterMPGDBarrelRecHits",
+            "BackwardMPGDEndcapRecHits",
+            "ForwardMPGDEndcapRecHits",
+            "B0TrackerRecHits"          // B0TRK
+        },
+        {"CentralTrackingRecHits"},
+        app));    // Output collection name
```
Comment (wdconinc) [inline src/global/tracking/tracking.cc:53]:
```suggestion
        {"CentralTrackingRecHits"}, // Output collection name
        app));
```

File: src/factories/tracking/TrackerHitCollector_factory.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
+// Created by Dmitry Romanov
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
```
Comment (wdconinc) [inline src/factories/tracking/TrackerHitCollector_factory.h:3]:
```suggestion
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright (C) 2023 Wouter Deconinck
```

File: src/factories/tracking/TrackerHitCollector_factory.cc [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// Created by Dmitry Romanov
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
```
Comment (wdconinc) [inline src/factories/tracking/TrackerHitCollector_factory.cc:3]:
```suggestion
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright (C) 2023 Wouter Deconinck
```

**Comment by wdconinc** [comment]
`{'set_item_removed': [root['CentralTrackingRecHits']], 'set_item_added': [root['CentralTrackingRecHits#0']]}` we are back to a proper subset collection, which is filled. Track seeding is completing as before as well.

Differences between https://pipelines.actions.githubusercontent.com/serviceHosts/4039b112-0c24-40a5-b9b2-2cdf69e0085f/_apis/pipelines/1/runs/3863/signedartifactscontent?artifactName=rec_dis_18x275_minQ2%3D1000_craterlake.edm4eic.root&urlExpires=2023-08-24T18%3A39%3A07.1839267Z&urlSigningMethod=HMACV2&urlSignature=8uOKyk4rp4UqHFznbSvhKihNVJLbR1ocXQ9ajKb22gY%3D and https://pipelines.actions.githubusercontent.com/serviceHosts/4039b112-0c24-40a5-b9b2-2cdf69e0085f/_apis/pipelines/1/runs/3847/signedartifactscontent?artifactName=rec_dis_18x275_minQ2%3D1000_craterlake.edm4eic.root&urlExpires=2023-08-24T18%3A39%3A49.9593948Z&urlSigningMethod=HMACV2&urlSignature=bGFg3ijyadpvvLEFUMNPftvnpOOWPRaj%2B%2FZqqUpVGn8%3D are overwhelmingly due to ordering of hits... sigh (I'm overwhelmed).

**Review by veprbl — APPROVED** [review]

**Pull request merged**